java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -3647,6 +3647,32 @@ public Object getCpu(User loggedInUser, Integer sid) {
             }
         }
 
+        if (ActionFactory.TYPE_PACKAGES_UPDATE.equals(acT)) {
+            List<Package> packages = packageMaps.stream().flatMap(packageMap -> {
+                PackageName packageName = PackageFactory.lookupPackageName(packageMap.get("name_id"));
+                PackageEvr evr = PackageEvrFactory.lookupPackageEvrById(packageMap.get("evr_id"));
+                PackageArch arch = PackageFactory.lookupPackageArchById(packageMap.get("arch_id"));
+
+                return PackageFactory.lookupByNevra(
+                        loggedInUser.getOrg(),
+                        packageName.getName(),
+                        evr.getVersion(),
+                        evr.getRelease(),
+                        evr.getEpoch(),
+                        arch
+                ).stream();
+            }).collect(toList());
+
+            List<Tuple2<Long, Long>> pidsidpairs = ErrataFactory.retractedPackages(
+                    packages.stream().map(p -> p.getId()).collect(toList()),
+                    sids.stream().map(s -> s.longValue()).collect(toList())
+            );
+            if (!pidsidpairs.isEmpty()) {
+                throw new RetractedPackageException(pidsidpairs.stream().map(t -> t.getA()).collect(toList()));
+            }
+        }
```

Comment:
Here the `suseServerChannelsRetractedPackagesView` could also be used. The best solution would probably be to use this view in the `Errata.retractedPackages` sql query.